### PR TITLE
Fixes #7570: If RUDDER_HOSTNAME is localhost we should fallback to FQDN

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/7570

### CODE REVIEW

**Comment by VinceMacBuche** [comment]
PR updated

File: inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala [hunk 1]
```diff
@@ -223,17 +222,20 @@ class FusionReportUnmarshaller(
...
 
-
   def processHostname (report : InventoryReport, xml : NodeSeq) : Box[InventoryReport] = {
 
     def updateHostname(hostname : String ) = {
       report.copy( node = report.node.copyWithMain(m => m.copy (hostname = hostname ) ))
     }
+
+    def invalidHostname( hostname : String ) : Boolean = {
+      hostname.isEmpty() && hostname == "localhost"
```
Comment (fanf) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: hostname can't be null ? (perhaps not)

Comment (fanf) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: and we don't check for 127.0.0.1 or ::1 ?

**Comment by VinceMacBuche** [comment]
PR updated

File: inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala [hunk 2]
```diff
@@ -223,17 +222,23 @@ class FusionReportUnmarshaller(
...
+  // Use RUDDER/HOSTNAME first and if missing OS/FQDN
   def processHostname (report : InventoryReport, xml : NodeSeq) : Box[InventoryReport] = {
 
     def updateHostname(hostname : String ) = {
       report.copy( node = report.node.copyWithMain(m => m.copy (hostname = hostname ) ))
     }
+
+    def invalidHostname( hostname : String ) : Boolean = {
+      val invalidList = "localhost" :: "127.0.0.1" :: "::1" :: Nil
+      hostname == null && hostname.isEmpty() && invalidList.contains(hostname)
```
Comment (fanf) [inline inventory-fusion/src/main/scala/com/normation/inventory/provisioning/fusion/FusionReportUnmarshaller.scala]: shouldn't be || ?

**Comment by VinceMacBuche** [comment]
PR updated

**Comment by VinceMacBuche** [comment]
PR updated

**Pull request merged**